tests/test_dulwich.py

```diff
@@ -26,3 +30,21 @@ def test_dulwich_github_compat(mocker: MockerFixture, algorithm: bytes):
     strings = iter((b"ssh-rsa", key_data))
     packet.get_string = lambda: next(strings)
     _process_public_key_ok_gh(auth, None, None, packet)
+
+
+def test_dulwich_get_ref(tmp_dir: TmpDir, scm: Git):
```

Comment:
Also, this test really belongs in the main test suite's `test_get_ref`. All 3 of the git backends support `get_ref`, and annotated tag support should not be dulwich specific, so we should test all of them.

Basically, the annotated tag specific part of this test should be added in:
https://github.com/iterative/scmrepo/blob/2561daf0915f5ed616fbaeed6e38c42f111ab354/tests/test_git.py#L205

(the other parts of this test are already implemented in the main `get_ref` test)

Reply:
Moved